Replace 'genetic locus' with SO gene

beckyjackson opened the pull request:
Resolves #129 

* Replaced 'genetic locus' (`REO:0000079`) with gene (`SO:0000704`)
* Updated all '... locus' labels to be '... gene'
* Obsoleted 'genetic entity' and moved haplotype, haplotype_block, and gene to top level

jamesaoverton commented:
I'm getting

```
ERROR: Validation completed with 387 message(s) at:
- 'genetic-locus' lines 3, 4, 5, ...
```

1. I think this validation script needs to be updated: https://github.com/IEDB/MRO/blob/master/src/scripts/validation/validate_templates.py#L314
2. I think Make should fail the build on these errors.

beckyjackson commented:
You're right - it wasn't failing because I had it set up for DROID, but I added a new command line option to get around this. I also updated the validation script to use the correct labels.
